package to demonstrate the features of movegroup interface

kruusamae opened the pull request:
the basic idea for this package is to write easy to understand coding examples to illustrate the capabilities of movegroup interface. for that reason:
1) the coding aims to be as minimalistic as possible with easy to understand variable names
2) different movegroup features are exemplified individually in different ROS nodes to make code graspable for beginners in computer programming

here the first two examples are provided: 1) planning to an end-effector pose and 2) planning to a named pose.

kruusamae committed:
pose_goal and named_goal nodes for review

kruusamae committed:
Updated README with initial instructions

kruusamae committed:
Minor fix in README

RValner commented on src/named_goal.cpp:
```diff
@@ -0,0 +1,77 @@
+
+  // MoveIt setup
+  // ^^^^^^^^^^^^
+  // A MoveGroupInterface instance (a client for the MoveGroup action) can be easily setup using just the name of the
```
Why are you referring to MoveGroup as action?

kruusamae replied:
mostly because of the constructor description reads, "Construct a client for the MoveGroup action for a particular group."
http://docs.ros.org/kinetic/api/moveit_ros_planning_interface/html/classmoveit_1_1planning__interface_1_1MoveGroupInterface.html

RValner reviewed (changes requested):
For the sake of giving a good example, please clean up the package.xml file from auto generated placeholders and unused blocks that are commented out.

Veix123 commented on src/named_goal.cpp:
```diff
@@ -0,0 +1,77 @@
+  else
+  {
+    ROS_WARN("[movegroup_interface_demo/named_goal] Planning failed. Shutting Down.");
+    ros::shutdown();
```
We should make sure the "execute plan" part is not reached in case of a planning failure... either move the execution part into the if(success) block above or simply
```suggestion
    ros::shutdown();
    return 0;
```

Veix123 commented on src/pose_goal.cpp:
```diff
@@ -0,0 +1,88 @@
+  else
+  {
+    ROS_WARN("[movegroup_interface_demo/pose_goal] Planning failed. Shutting Down.");
+    ros::shutdown();
```
```suggestion
    ros::shutdown();
    return 0;
```

Veix123 commented on CMakeLists.txt:
```diff
@@ -0,0 +1,51 @@
+catkin_package(
+#  INCLUDE_DIRS include
+#  LIBRARIES movegroup_interface_tutorial
+#  CATKIN_DEPENDS geometry_msgs roscpp rospy std_msgs
```
Need to declare catkin dependencies for other packages which may include this one. It seems that only geometry_msgs are used in this package.
```suggestion
  CATKIN_DEPENDS geometry_msgs roscpp
```

Veix123 reviewed (changes requested):
+1 for cleaning up package.xml, also fill in the missing author and license tags.
if this package demonstrates only cpp examples I would remove python dependencies from package.xml and CMakeLists.txt.

kruusamae committed:
removed redundant dependencies from CMakeLists, added proper includes to .cpp files

kruusamae committed:
cleaned up package.xml

kruusamae committed:
making sure the 'execute plan' part is not reached in case of a planning failure

kruusamae commented:
here's my changes related to your feedback.
1) cleaned up CMakeLists.txt as suggested by @Veix123 
2) cleaned up package.xml as suggested by @RValner . Please double-check the contents, I've never been completely sure how to properly define package.xml
3) added "return 0;" as suggested by @Veix123 
4) added new includes, i hope it makes the code files a bit better quality; fixed a mistake in named_goal node name in ros::init()

kruusamae committed:
added moveit_ros_planning_interface as a dependency
